dbt-bigquery/src/dbt/include/bigquery/macros/materializations/incremental_strategy/insert_overwrite.sql

```diff
@@ -40,62 +40,144 @@
+    -- with `insert_overwrite`, dbt uses a predefined partition filter
+    -- expression to overwrite specific partitions
+    -- (e.g., where partition_column = '2024-05-01').
+    -- it excels where using a fixed minimal partition list, no need to
+    -- query source data for partitions, and insert statements can be
+    -- batched. this also keeps things deterministic.
+#}
+
+{% macro bq_static_select_insert_overwrite_sql(tmp_relation, sql, partition_by, tmp_relation_exists) %}
+  {%- set source_sql -%}
+  (
+    {% if tmp_relation_exists %}
+      select
+        {% if partition_by.time_ingestion_partitioning %}
+          {{ partition_by.insertable_time_partitioning_field() }},
+        {% endif %}
+        * from {{ tmp_relation }}
+    {%- elif partition_by.time_ingestion_partitioning -%}
+      {{ wrap_with_time_ingestion_partitioning_sql(partition_by, sql, true) }}
+    {%- else -%}
+      {{ sql }}
+    {%- endif %}
+  )
+  {%- endset -%}
+  {{ return(source_sql) }}
+{% endmacro %}
+
+
+{#
+  -- Static-partition + `copy_partitions=true` should inline the literals / foldable constants
+  -- supplied via the `partitions=` config.
+
+  -- The inline method will still copy (truncate) a partition even
+  -- if the incremental run produced no rows for that date, because the user
+  -- explicitly listed it in `partitions=`.
+#}
+{% macro bq_static_copy_partitions_insert_overwrite_sql(
+  tmp_relation, target_relation, sql, partition_by, partitions, tmp_relation_exists
 ) %}
 
-      {% set predicate -%}
-          {{ partition_by.render_wrapped(alias='DBT_INTERNAL_DEST') }} in (
-              {{ partitions | join (', ') }}
-          )
-      {%- endset %}
+    {%- if tmp_relation_exists is false -%}
+        {%- set source_sql = bq_static_select_insert_overwrite_sql(tmp_relation, sql, partition_by, tmp_relation_exists) %}
 
-      {%- set source_sql -%}
-        (
-          {% if partition_by.time_ingestion_partitioning and tmp_relation_exists -%}
-          select
-            {{ partition_by.insertable_time_partitioning_field() }},
-            * from {{ tmp_relation }}
-          {% elif tmp_relation_exists -%}
-            select
-            * from {{ tmp_relation }}
-          {%- elif partition_by.time_ingestion_partitioning -%}
-            {{ wrap_with_time_ingestion_partitioning_sql(partition_by, sql, True) }}
-          {%- else -%}
-            {{sql}}
-          {%- endif %}
-
-        )
-      {%- endset -%}
+        {# -- we run temp table creation in a separate script to move to partitions copy if it doesn't already exist #}
+        {%- call statement('create_tmp_relation_for_copy', language='sql') -%}
+          {{ bq_create_table_as(partition_by, true, tmp_relation, source_sql, 'sql')
+        }}
+        {%- endcall %}
+    {%- endif -%}
 
-      {% if copy_partitions %}
-          {% do bq_copy_partitions(tmp_relation, target_relation, partitions, partition_by) %}
-      {% else %}
+    {%- set partitions_sql -%}
```

Comment:
I changed this up from Chris' original PR using a bigquery sql trick to inline the table after finding a bug with multi-literal partition lists.

Reply:
I might be confused but this is calling out to bigquery to convert `partition_literal` (strings) into bigquery timestamps? Could we do this processing locally?

Reply:
Code style of other branches, didn't want to make a whole method, but now I know about `@parse.none`. It's coming right up